tests/Unit/DataStructures/Tensor/Expressions/Test_Product.cpp

```diff
@@ -0,0 +1,1099 @@
+          TensorExpressions::evaluate<ti_i, ti_a>(R() * Tll(ti_a, ti_i));
+  // \f$L_{ai} = T_{ai} * R\f$
+  const Tensor<DataType, Symmetry<2, 1>,
+               index_list<SpacetimeIndex<3, UpLo::Lo, Frame::Inertial>,
+                          SpatialIndex<4, UpLo::Lo, Frame::Inertial>>>
+      Lai_from_Tai_R =
+          TensorExpressions::evaluate<ti_a, ti_i>(Tll(ti_a, ti_i) * R());
+  // \f$L_{ia} = T_{ai} * R\f$
+  const Tensor<DataType, Symmetry<2, 1>,
+               index_list<SpatialIndex<4, UpLo::Lo, Frame::Inertial>,
+                          SpacetimeIndex<3, UpLo::Lo, Frame::Inertial>>>
+      Lia_from_Tai_R =
+          TensorExpressions::evaluate<ti_i, ti_a>(Tll(ti_a, ti_i) * R());
+
+  for (size_t a = 0; a < 4; a++) {
+    for (size_t i = 0; i < 4; i++) {
+      const DataType expected_R_Tai_product = R.get() * Tll.get(a, i);
+      CHECK(Lai_from_R_Tai.get(a, i) == expected_R_Tai_product);
+      CHECK(Lia_from_R_Tai.get(i, a) == expected_R_Tai_product);
+
+      const DataType expected_Tai_R_product = Tll.get(a, i) * R.get();
+      CHECK(Lai_from_Tai_R.get(a, i) == expected_Tai_R_product);
+      CHECK(Lia_from_Tai_R.get(i, a) == expected_Tai_R_product);
+    }
+  }
+}
+
+/// \ingroup TestingFrameworkGroup
+/// \brief Test the outer product of rank 1 tensors with another tensor is
+/// correctly evaluated
+///
+/// \details
+/// The outer product cases tested are:
+/// - \f$L^{a}{}_{i} = R_{i} * S^{a}\f$
+/// - \f$L_{ja}{}^{i} = R_{i} * S^{a} * T^{j}\f$
+/// - \f$L_{k}{}^{c}{}_{d} = S^{c} * G_{dk}\f$
+/// - \f$L^{c}{}_{dk} = G_{dk} * S^{c}\f$
+///
+/// For each case, the LHS index order is different from the order in the
+/// operands.
+///
+/// \tparam DataType the type of data being stored in the product operands
+template <typename DataType>
+void test_outer_product_rank_1_operand(const DataType& used_for_size) noexcept {
+  Tensor<DataType, Symmetry<1>,
+         index_list<SpatialIndex<3, UpLo::Lo, Frame::Grid>>>
+      Rl(used_for_size);
+  create_tensor(make_not_null(&Rl));
+
+  Tensor<DataType, Symmetry<1>,
+         index_list<SpacetimeIndex<3, UpLo::Up, Frame::Grid>>>
+      Su(used_for_size);
+  create_tensor(make_not_null(&Su));
+
+  // \f$L^{a}{}_{i} = R_{i} * S^{a}\f$
+  // Use explicit type (vs auto) for LHS Tensor so the compiler checks the
+  // return type of `evaluate`
+  const Tensor<DataType, Symmetry<2, 1>,
+               index_list<SpacetimeIndex<3, UpLo::Up, Frame::Grid>,
+                          SpatialIndex<3, UpLo::Lo, Frame::Grid>>>
+      LAi_from_Ri_SA =
+          TensorExpressions::evaluate<ti_A, ti_i>(Rl(ti_i) * Su(ti_A));
+
+  for (size_t i = 0; i < 3; i++) {
+    for (size_t a = 0; a < 4; a++) {
+      CHECK(LAi_from_Ri_SA.get(a, i) == Rl.get(i) * Su.get(a));
+    }
+  }
+
+  Tensor<DataType, Symmetry<1>,
+         index_list<SpatialIndex<3, UpLo::Up, Frame::Grid>>>
+      Tu(used_for_size);
+  create_tensor(make_not_null(&Tu));
+
+  // \f$L_{ja}{}^{i} = R_{i} * S^{a} * T^{j}\f$
```

Comment:
Same comment about up/down...

Based on the code below, it looks like the RHS is correct? So I think the comment should change but everything else is correct?